ntoskrnl/mm/ARM3/pagfault.c

```diff
@@ -76,27 +76,46 @@ MiCheckForUserStackOverflow(IN PVOID Address,
     }
 
     /* This is where the stack will start now */
-    NextStackAddress = (PVOID)((ULONG_PTR)PAGE_ALIGN(Address) - GuranteedSize);
+    NextStackAddress = (PVOID)((ULONG_PTR)PAGE_ALIGN(Address) - GuaranteedSize);
+
+    /* Don't handle this flag yet */
+    ASSERT((PsGetCurrentProcess()->Peb->NtGlobalFlag & FLG_DISABLE_STACK_EXTENSION) == 0);
 
     /* Do we have at least one page between here and the end of the stack? */
     if (((ULONG_PTR)NextStackAddress - PAGE_SIZE) <= (ULONG_PTR)DeallocationStack)
     {
-        /* We don't -- Windows would try to make this guard page valid now */
-        DPRINT1("Close to our death...\n");
+        /* We don't -- try making this guard page valid
+         * by commiting the page before the last page of the stack */
+
+        NextStackAddress = (PVOID)((ULONG_PTR)PAGE_ALIGN(DeallocationStack) + GuaranteedSize);
+
+        Status = ZwAllocateVirtualMemory(NtCurrentProcess(),
+            &NextStackAddress,
+            0,
+            &GuaranteedSize,
+            MEM_COMMIT,
+            PAGE_READWRITE);
+
+        if (NT_SUCCESS(Status))
+        {
+            /* Update the stack limit */
+            Teb->NtTib.StackLimit = (PVOID)(ULONG_PTR)NextStackAddress;
+
+            /* We did it! */
+            DPRINT1("The last guard page handled successfully for %p\n", Address);
+        }
+
         return STATUS_STACK_OVERFLOW;
     }
 
-    /* Don't handle this flag yet */
-    ASSERT((PsGetCurrentProcess()->Peb->NtGlobalFlag & FLG_DISABLE_STACK_EXTENSION) == 0);
```

Comment:
Why does this disappear?

Reply:
It doesn't. It has moved. But I will revert this functional change in favor of typo-fixing one only.

Reply:
@andrew-boyarshin, any ETA on the change you proposed? This will avoid holding this PR forever.
Otherwise, it'll just get closed.

Reply:
@HeisSpiter I'm working on Vista+ LDR right now (a11y branch already has stable and working TLS support for dynamically loaded libraries, .NET image mscoree support in the works). But yeah, I could update this PR (since a11y already has it done). ETA: 10 hours.

Reply:
@HeisSpiter after having a look at the state of this PR - it's already done 16 days ago. This is the only functional change left (since I believed PR 1035 missed this one), as I intended it to be. But now I'm not sure if it's reasonable to retain this change. Reverted this.